autosklearn/data/feature_validator.py

```diff
@@ -0,0 +1,399 @@
+                for column in X.columns:
+                    if X[column].isna().all():
+                        X[column] = pd.to_numeric(X[column])
+
+            self.enc_columns, self.feature_types = self._get_columns_to_encode(X)
+
+            if len(self.enc_columns) > 0:
+
+                self.encoder = make_column_transformer(
+                    (preprocessing.OrdinalEncoder(), self.enc_columns),
+                    remainder="passthrough"
+                )
+
+                # Mypy redefinition
+                assert self.encoder is not None
+                self.encoder.fit(X)
+
+                # The column transformer reoders the feature types - we therefore need to change
+                # it as well
+                def comparator(cmp1: str, cmp2: str) -> int:
+                    if (
+                        cmp1 == 'categorical' and cmp2 == 'categorical'
+                        or cmp1 == 'numerical' and cmp2 == 'numerical'
+                    ):
+                        return 0
+                    elif cmp1 == 'categorical' and cmp2 == 'numerical':
+                        return -1
+                    elif cmp1 == 'numerical' and cmp2 == 'categorical':
+                        return 1
+                    else:
+                        raise ValueError((cmp1, cmp2))
+                self.feature_types = sorted(
+                    self.feature_types,
+                    key=functools.cmp_to_key(comparator)
+                )
+        return self
+
+    def transform(
+        self,
+        X: typing.Union[np.ndarray, pd.DataFrame],
+    ) -> np.ndarray:
+        """
+        Validates and fit a categorical encoder (if needed) to the features.
+        The supported data types are List, numpy arrays and pandas DataFrames.
+
+        Parameters
+        ----------
+            X_train: typing.Union[np.ndarray, pd.DataFrame]
+                A set of features, whose categorical features are going to be
+                transformed
+
+        Return
+        ------
+            np.ndarray:
+                The transformed array
+        """
+        if not self._is_fitted:
+            raise ValueError("Cannot call transform on a validator that is not fitted")
+
+        # Pandas related transformations
+        if hasattr(X, "iloc") and self.encoder is not None:
+            if np.any(pd.isnull(X)):
+                # After above check it means that if there is a NaN
+                # the whole column must be NaN
+                # Make sure it is numerical and let the pipeline handle it
+                for column in X.columns:
+                    if X[column].isna().all():
+                        X[column] = pd.to_numeric(X[column])
+            try:
+                X = self.encoder.transform(X)
+            except ValueError as e:
+                if 'Found unknown categories' in e.args[0]:
+                    # Make the message more informative
+                    raise ValueError(
```

Comment:
Can this error still happen with sklearn 0.24? I thought that they added a passthrough for new categories, probably this needs to be manually turned on?

Reply:
Thanks, I did not know about this! Yes, one has to manually enable this and I have done so now.